For mozilla-mobile/fenix#8014 - fix feature-pwa manifests db migration

TitanNano opened the pull request:
We need to increase the db version so room creates a new schema and migrate the schema from the original version to the new one.
---
<!-- Text above this line will be added to the commit once "bors" merges this PR -->

### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

NotWoods commented on components/feature/pwa/build.gradle:
```diff
@@ -12,6 +12,7 @@ android {
     defaultConfig {
         minSdkVersion config.minSdkVersion
         targetSdkVersion config.targetSdkVersion
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Is this necessary? Most of our tests specify the runner with `RunWith`

TitanNano replied:
I'm actually not sure. Will check it out.

TitanNano replied:
It seems I can't use `RunWith`: `Type mismatch: inferred type is KClass<AndroidJUnitRunner> but KClass<out Runner!> was expected`. 

I borrowed this from https://github.com/mozilla-mobile/android-components/blob/f7d2a4648162bebc98f852e2f2f9d6bd8b6f0636/components/browser/engine-gecko-nightly/build.gradle#L30 where it's also done like this.

TitanNano commented:
@NotWoods in my initial patch I reverted schema version 1 to before #5724 and migrated from there. But I think  that would cause issues for people who initially got the new version 1, so I added a version 0 which is the original version 1 and then test the migration against both version 1 to 2 and version 0 to 2. I hope this smoothens things out as good as possible.

NotWoods commented:
bors r=NotWoods

TitanNano commented:
@NotWoods whups, there are still two unused imports I missed, can I still push an update for this?

NotWoods commented:
Yes, pushing will cancel the merge. It won't automerge if linting fails anyways

NotWoods commented:
bors r=NotWoods

NotWoods commented:
bors retry
